kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigSecret.java

```diff
@@ -205,22 +208,30 @@ private static GenericKubernetesResource filterResourceFromRef(Clusters cluster,
                 .orElseThrow(() -> resourceNotFound(cluster, filterRef));
     }
 
-    private static VirtualCluster getVirtualCluster(KafkaProxy primary,
-                                                    Clusters cluster) {
-        var o = ResourcesUtil.distinctClusters(primary);
-        var clusterNum = o.indexOf(cluster);
+    private record NamedVirtualCluster(String name, VirtualCluster virtualCluster) {
+
+    }
 
+    private static Stream<NamedVirtualCluster> getNamedVirtualClusters(KafkaProxy primary, Clusters cluster) {
```

Comment:
This method appears to have a single call site which basically converts the `Stream<NamedVirtualCluster>` into a `Map` keyed on the cluster name. So maybe just make this method return that `Map`?